File: ios/JJMproject/Classes/JSBridge/Managers/Dialog/DialogModule.m
//
//  NetworkModule.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/14.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DialogModule, NSObject)

RCT_EXTERN_METHOD(showSuccessWithAction:(NSString *)message cancelable:(BOOL)cancelable positiveTitle:(NSString *)positiveTitle negativeTitle:(NSString *)negativeTitle positiveAction:(RCTResponseSenderBlock)positiveAction)
RCT_EXTERN_METHOD(showSuccessWithActions:(NSString *)message cancelable:(BOOL)cancelable positiveTitle:(NSString *)positiveTitle negativeTitle:(NSString *)negativeTitle positiveAction:(RCTResponseSenderBlock)positiveAction negativeAction:(RCTResponseErrorBlock)negativeAction)

RCT_EXTERN_METHOD(showFailureWithAction:(NSString *)message cancelable:(BOOL)cancelable positiveTitle:(NSString *)positiveTitle negativeTitle:(NSString *)negativeTitle positiveAction:(RCTResponseSenderBlock)positiveAction)
RCT_EXTERN_METHOD(showFailureWithActions:(NSString *)message cancelable:(BOOL)cancelable positiveTitle:(NSString *)positiveTitle negativeTitle:(NSString *)negativeTitle positiveAction:(RCTResponseSenderBlock)positiveAction negativeAction:(RCTResponseErrorBlock)negativeAction)

RCT_EXTERN_METHOD(showMessageWithAction:(NSString *)message cancelable:(BOOL)cancelable positiveTitle:(NSString *)positiveTitle negativeTitle:(NSString *)negativeTitle positiveAction:(RCTResponseSenderBlock)positiveAction)
RCT_EXTERN_METHOD(showMessageWithActions:(NSString *)message cancelable:(BOOL)cancelable positiveTitle:(NSString *)positiveTitle negativeTitle:(NSString *)negativeTitle positiveAction:(RCTResponseSenderBlock)positiveAction negativeAction:(RCTResponseErrorBlock)negativeAction)
RCT_EXTERN_METHOD(dismiss)

@end

File: ios/JJMproject/Classes/JSBridge/Managers/Dialog/DialogModule.swift
//
//  DialogModule.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/12/12.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import UIKit
import Async

@objc(DialogModule)
class DialogModule: NSObject {
  
  private lazy var dialogManager: DialogManager = {
    return DialogManager()
  }()
  
  @objc func showSuccessWithAction(_ message: String, cancelable: Bool, positiveTitle: String, negativeTitle: String, positiveAction: @escaping RCTResponseSenderBlock) {
    Async.main {
//      let dialogManager = DialogManager()
      self.dialogManager.showSuccess(message: message, positiveTitle: positiveTitle) {
        self.dialogManager.dismiss()
        positiveAction(nil)
      }
    }
  }
  
  @objc func showSuccessWithActions(_ message: String, cancelable: Bool, positiveTitle: String, negativeTitle: String, positiveAction: @escaping RCTResponseSenderBlock, negativeAction: @escaping RCTResponseErrorBlock) {
    Async.main {
//      let dialogManager = DialogManager()
      self.dialogManager.showSuccess(message: message, positiveTitle: positiveTitle, negativeTitle: negativeTitle, positiveHandler: {
        self.dialogManager.dismiss()
        positiveAction(nil)
      }) {
        self.dialogManager.dismiss()
        negativeAction(nil)
      }
    }
  }
  
  @objc func showFailureWithAction(_ message: String, cancelable: Bool, positiveTitle: String, negativeTitle: String, positiveAction: @escaping RCTResponseSenderBlock) {
    Async.main {
//      let dialogManager = DialogManager()
      self.dialogManager.showFailure(message: message, positiveTitle: positiveTitle) {
        self.dialogManager.dismiss()
        positiveAction(nil)
      }
    }
  }
  
  @objc func showFailureWithActions(_ message: String, cancelable: Bool, positiveTitle: String, negativeTitle: String, positiveAction: @escaping RCTResponseSenderBlock, negativeAction: @escaping RCTResponseErrorBlock) {
    Async.main {
//      let dialogManager = DialogManager()
      self.dialogManager.showFailure(message: message, positiveTitle: positiveTitle, negativeTitle: negativeTitle, positiveHandler: {
        self.dialogManager.dismiss()
        positiveAction(nil)
      }) {
        self.dialogManager.dismiss()
        negativeAction(nil)
      }
    }
  }
  
  @objc func showMessageWithAction(_ message: String, cancelable: Bool, positiveTitle: String, negativeTitle: String, positiveAction: @escaping RCTResponseSenderBlock) {
    Async.main {
//      let dialogManager = DialogManager()
      self.dialogManager.showWarning(message: message, positiveTitle: positiveTitle) {
        self.dialogManager.dismiss()
        positiveAction(nil)
      }
    }
  }
  
  @objc func showMessageWithActions(_ message: String, cancelable: Bool, positiveTitle: String, negativeTitle: String, positiveAction: @escaping RCTResponseSenderBlock, negativeAction: @escaping RCTResponseErrorBlock) {
    Async.main {
//      let dialogManager = DialogManager()
      self.dialogManager.showWarning(message: message, positiveTitle: positiveTitle, negativeTitle: negativeTitle, positiveHandler: {
        self.dialogManager.dismiss()
        positiveAction(nil)
      }) {
        self.dialogManager.dismiss()
        negativeAction(nil)
      }
    }
  }
  
  @objc func dismiss() {
    Async.main {
      self.dialogManager.dismiss()
    }
  }

}
